### DISCUSSION TITLE
How can a Github workflow read a file on another private repository hosted on the same GitHub account?

### DISCUSSION THREAD

**Discussion by AllanXu49** [community/community#27201] [Actions]
<p>I have a <strong>private</strong> git repository like:</p>
<p><code>Github.com/acct/repo1</code></p>
<p>I have a <code>test.json</code> file in another private repository in the same GitHub subscription and account like:</p>
<p><code>Github.com/acct/repo2/test.json</code></p>
<p>How can a workflow in <code>repo1</code> have http get access to the url <code>Github.com/acct/repo2/test.json</code> ?</p>
<p>All I need is a simple way to read the content of <code>test.json</code> in a workflow running on <code>repo2</code> .</p>
<p>It works the best for me to read the file using a curl get request.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You can use a <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token">Personal Access Token</a> (PAT) with the <code>repo</code> scope to authorize the request to the <code>raw.githubusercontent.com</code> URL for the file. If you add the PAT as a <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token">secret</a> for your repository, you can do something like this:</p>
<pre><code class="lang-yaml">- name: get file
  env:
    TOKEN: ${{ secrets.repo_token }}
  run: |
    curl -H "Authorization: token ${TOKEN}" https://raw.githubusercontent.com/acct/repo2/main/test.json
</code></pre>

**Comment by AllanXu49** [comment]
<p>Thank you so much for help.</p>
<p>Any suggestion for learning path where do I learn details like this?</p>

**Comment by airtower-luna** [comment]
<p>You’re welcome! :smiley_cat:</p>
<p>Nothing too specific, I’m afraid. The key here is knowing that PATs can be used to authorize all sorts of things. I then just checked the <a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps">documentation on scopes</a> to see which would be needed, and tried downloading a file from a private repository to make sure it’d actually work. I picked up most of the general concepts by poking around the documentation and reading here on the forum. :wink:</p>
